sympy/combinatorics/tests/test_homomorphisms.py

```diff
@@ -55,3 +55,33 @@ def test_homomorphism():
     assert T.domain == F
     assert T.codomain == D
     assert T(a*b) == p
+
+def test_isomorphisms():
+    # Trivial Case
+    # FpGroup -> FpGroup
+    F, a, b = free_group("a, b")
+    H = FpGroup(F, [a**3, b**3, (a*b)**2])
+    F, c, d = free_group("c, d")
+    G = FpGroup(F, [c**3, d**3, (c*d)**2])
+    assert is_isomorphism(G, H)
+
+    # FpGroup -> PermutationGroup
+    # FpGroup is converted to the equivalent isomorphic group.
+    F, a, b = free_group("a, b")
+    G = FpGroup(F, [a**3, b**3, (a*b)**2])
+    H = AlternatingGroup(4)
+    T = group_isomorphism(G, H)
+    assert T.kernel().order() == G.order()
+    assert T.is_trivial()
+    assert is_isomorphism(G, H)
+
+    # PermutationGroup -> PermutationGroup
+    D = DihedralGroup(8)
+    p = Permutation(0, 1, 2, 3, 4, 5, 6, 7)
+    P = PermutationGroup(p)
+    assert not is_isomorphism(D, P)
+
+    # Cyclic Groups of prime order are isomorphic to each other.
```

Comment:
No, they aren't. All group of prime order are cyclic, so two groups of the same prime order are isomorphic. But The groups you have below are certainly not isomorphic: one has order 5 and the other 7 - so this test shouldn't pass.

Reply:
The problem here might be because of the `PermutationGroup`. While mapping from the generators of G to the elements of H, we just check if the length of the generators is equal to the size of the subset. For example, in this case: 
```
>>> A = CyclicGroup(5)
>>> B = CyclicGroup(7)
>>> T = group_isomorphism(A, B)
>>> T
{Permutation(0, 1, 2, 3, 4): Permutation(0, 1, 2, 3, 4, 5, 6)}
``` 
The map is actually between 2 permutations of different size. 
So I guess checking the size of the permutation before we map could solve it, will that do?

Reply:
That should definitely be checked. But firstly, I'm uncertain if that will be sufficient for all cases, and secondly, it surprises me that `homomorphism` allows this - it was designed to return an error if a given mapping doesn't extend to a valid homomorphism. So this is possibly a bug. I'll look into it.